// Copyright (c) Authors of Clover
//
// All rights reserved. This program and the accompanying materials
// are made available under the terms of the Apache License, Version 2.0
// which accompanies this distribution, and is available at
// http://www.apache.org/licenses/LICENSE-2.0

syntax = "proto3";

package jmeter;

// The controller service definition.
service Controller {

  rpc GenTest (ConfigJmeter) returns (JmeterReply) {}
  rpc StartTest (TestParams) returns (JmeterReply) {}
  rpc GetResults (JResults) returns (JmeterReply) {}
}

message TestParams {
  string num_slaves = 1;
  string test_plan = 2;
  string slave_ips = 3;
}

message ConfigJmeter {
  string url_list = 1;
  string num_threads = 2;
  string url_names = 3;
  string url_protocols = 4;
  string url_methods = 5;
  string loops = 6;
  string ramp_time = 7;
}

message JmeterReply {
  string message = 1;
}

message JResults {
  string r_format = 1;
  string r_file = 2;
}
